test/connectivity/connectivity.test.jsx

```diff
@@ -0,0 +1,259 @@
+const React = require('react');
+const debug = require('debug')('connectivity-test');
+const { promisify } = require('util');
+
+Enzyme.configure({ adapter: new Adapter() });
+const { mount } = Enzyme;
+
+// Stub electron module for tests
+const originalLoader = m._load;
+const stubs = {
+  electron: {
+    ipcMain: {
+      call: (methodName) => {
+        debug('electron.ipcMain.call main called!', methodName);
+      },
+      respondTo: (methodName) => {
+        debug('electron.ipcMain.respondTo main called!', methodName);
+      },
+      on: (methodName) => {
+        debug('electron.ipcMain.on main called!', methodName);
+      }
+    },
+    remote: {
+      app: {
+        getName: () => 'Compass Connectivity Integration Test Suite',
+        getPath: () => ''
+      },
+      dialog: {
+        remote: {},
+        BrowserWindow: {}
+      }
+    },
+    shell: {}
+  }
+};
+m._load = function hookedLoader(request, parent, isMain) {
+  const stub = stubs[request];
+  return stub || originalLoader(request, parent, isMain);
+};
+
+// Mock dom/window for tests.
+// Has to come before @mongodb-js/compass-connect require.
+jsdomGlobal();
+
+const AppRegistry = require('hadron-app-registry');
+const Connection = require('mongodb-connection-model');
+const CompassConnectPlugin = require('@mongodb-js/compass-connect');
+const CompassConnectComponent = CompassConnectPlugin.default;
+const activateCompassConnect = CompassConnectPlugin.activate;
+
+const deactivateCompassConnect = CompassConnectPlugin.deactivate;
+
+const TEST_TIMEOUT_MS = require('./.mocharc.json').timeout;
+const {
+  dockerComposeDown,
+  dockerComposeUp
+} = require('./docker-instance-manager');
+
+// Hide react warnings.
+const originalWarn = console.warn.bind(console.warn);
+console.warn = (msg) => (
+  !msg.toString().includes('componentWillReceiveProps')
+  && !msg.toString().includes('componentWillUpdate')
+  && originalWarn(msg)
+);
+
+const delay = promisify(setTimeout);
+const ensureConnected = async(timeout, testIsConnected) => {
+  let connected = await testIsConnected();
+  let timespentTesting = 0;
+  while (!connected) {
+    if (timeout > TEST_TIMEOUT_MS || timespentTesting > TEST_TIMEOUT_MS) {
+      throw new Error('Waited for connection, never happened');
+    }
+    debug(`Testing connectivity at timeout=${timeout}, connected=${connected}`);
+    await delay(timeout);
+    timespentTesting += timeout;
+    timeout *= 2; // Try again but wait double.
+    connected = await testIsConnected();
+  }
+  return connected;
+};
+
+const connectionsToTest = [{
+  title: 'local community standalone',
+  dockerComposeFilePath: 'community/docker-compose.yaml',
+  model: {
+    hostname: 'localhost',
+    port: 27020,
+  },
+  expectedInstanceDetails: {
+    _id: 'localhost:27020',
+    hostname: 'localhost',
+    port: 27020,
+    client: {
+      isWritable: true,
+      isMongos: false,
+    },
+    build: {
+      raw: {
+        version: /4.4.[1-9]+$/
```

Comment:
This version check will error when there's a minor version update - I'll update to only check for major updates 🤔 or maybe any version that is a valid version